src/bitops.c

```diff
@@ -897,13 +903,7 @@ void bitposCommand(client *c) {
         return;
     }
 
-    /* If the key does not exist, from our point of view it is an infinite
-     * array of 0 bits. If the user is looking for the first clear bit return 0,
-     * If the user is looking for the first set bit, return -1. */
-    if ((o = lookupKeyRead(c->db,c->argv[1])) == NULL) {
-        addReplyLongLong(c, bit ? -1 : 0);
-        return;
-    }
+    o = lookupKeyRead(c->db, c->argv[1]);
     if (checkType(c,o,OBJ_STRING)) return;
```

Comment:
i now realize that the type check in BITPOS is still ahead of the syntax error

Reply:
right, i just realize it, i am making the changes. (been looking at BITCOUNT before)